### PR TITLE
Create global variables from PDB

### LINKED ISSUE #3625: Create global variables from debug information (DWARF, PDB, etc)
Now that we removed typelinks and added types propagation from the global variables (see `aat` and `avg` commands), they also should be created when the information is available in DWARF, PDB, and any other debug info information format.
- [x] DWARF
- [x] PDB

### PR DESCRIPTION
**Pull request by b1llow**
**Your checklist for this pull request**
- [ ] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository
- [ ] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style)
- [ ] I've documented or updated the documentation of every function and struct this PR changes. If not so I've explained why.
- [ ] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the [rizin book](https://github.com/rizinorg/book) with the relevant information (if needed)

**Detailed description**

...

**Test plan**

...

**Closing issues**

closes #3625

### CODE REVIEW

File: librz/bin/pdb/omap.c [hunk 1]
```diff
@@ -88,21 +71,19 @@ static int binary_search(unsigned int *A, int key, int imin, int imax) {
  * \param address Where to remap
  * \return int
  */
-RZ_API int rz_bin_pdb_omap_remap(RZ_NONNULL RzPdbOmapStream *omap_stream, int address) {
+static ut64 rz_bin_pdb_omap_remap(RZ_NONNULL RzPdbOmapStream *omap_stream, ut64 address) {
```
Comment (XVilka) [inline librz/bin/pdb/omap.c]: If it's static, please remove `rz_` prefix

File: librz/bin/pdb/omap.c [hunk 2]
```diff
@@ -133,3 +109,17 @@ RZ_API int rz_bin_pdb_omap_remap(RZ_NONNULL RzPdbOmapStream *omap_stream, int ad
 	}
 	return omap_entry->to + (address - omap_entry->from);
 }
+
+RZ_API ut64 rz_bin_pdb_to_rva(RZ_NONNULL const RzPdb *pdb, const PDBSectionOffset *section_offset) {
```
Comment (XVilka) [inline librz/bin/pdb/omap.c:120]: Please add Doxygen

File: librz/bin/pdb/tpi.c [hunk 3]
```diff
@@ -403,34 +444,34 @@ RZ_API bool rz_bin_pdb_type_is_fwdref(RZ_NONNULL RzPdbTpiType *t) {
  * \param t RzPdbTpiType
  * \return RzList *
```
Comment (XVilka) [inline librz/bin/pdb/tpi.c]: Update Doxygen too

File: librz/include/rz_util/rz_buf.h [hunk 4]
```diff
@@ -349,6 +350,18 @@ DEFINE_RZ_BUF_WRITE_OFFSET_BLE(64)
 #undef DEFINE_RZ_BUF_READ_OFFSET_BLE
 #undef DEFINE_RZ_BUF_WRITE_OFFSET_BLE
 
+static inline ut8 rz_buf_peek(RZ_NONNULL RzBuffer *b) {
```
Comment (XVilka) [inline librz/include/rz_util/rz_buf.h:361]: Please add Doxygen for those

File: test/db/formats/pdb [hunk 5]
```diff
@@ -1,3 +1,40 @@
+NAME=Load global variables from PDB
+FILE=
+CMDS=<<EOF
+idp bins/pdb/Project1.pdb
+avg
+EOF
+EXPECT=<<EOF
```
Comment (XVilka) [inline test/db/formats/pdb:7]: Could you also please add a test that checks if those (just a few, at least two) global variables are referenced? E.g. using both `avgx` and `pd`/`pdf` commands?

Comment (XVilka) [reply test/db/formats/pdb:7]: This way, we make sure that these globals are picked up by the analysis.

**Review by XVilka — CHANGES_REQUESTED** [review]
rz-bindgen failure:
```
Run python3 rz-bindgen/src/lint.py \
Type comment at <rizin/librz/bin/pdb/tpi.h:543:24> lacks angle brackets
Error: Process completed with exit code 1.
```

I think it might make sense to try fixing also https://github.com/rizinorg/rizin/issues/955 in the same PR (if it's easy enough, of course)

File: librz/bin/pdb/tpi.h [hunk 6]
```diff
@@ -487,6 +534,22 @@ typedef struct {
 	ut8 pad;
 } Tpi_LF_Vtshape;
 
+/// type record for a virtual function table
+typedef struct {
+	ut16 leaf; // LF_VFTABLE
+	TpiCVType complete_class; // class/structure that owns the vftable
+	TpiCVType override_vftable; // vftable from which this vftable is derived
+	ut32 vfptr_offset; // offset of the vfptr to this table, relative to the start of the object layout.
+	RzPVector /* <char *> */ method_names; // array of names.
```
Comment (wargio) [inline librz/bin/pdb/tpi.h]:
```suggestion
	RzPVector /*<char *>*/ method_names; // array of names.
```

**Pull request merged**